### PR TITLE
Return undefined when query param is not provided (i.e. Zod optional params)

### PR DESCRIPTION
**Pull request by kylem123**
Encountered issue when using optional or nullish props in Zod schema. Essentially, the get query param function will attempt to fetch these props, but return an empty array. This then causes the zod schema to error, as it is expecting string.optional for example, but it receives an array. This PR alters this behaviour to return undefined in the case that the array is empty.

### CODE REVIEW

**Comment by ardatan** [comment]
Thanks for the PR!
Could you add a changeset for a release(using `yarn changeset`) and a test as well to prevent future regressions?

File: packages/fets/src/createRouter.ts [hunk 1]
```diff
@@ -153,6 +153,9 @@ export function createRouterBase(
           {
             get(_, prop) {
               const allQueries = url.searchParams.getAll(prop.toString());
+              if (prop !== "then" && allQueries.length === 0) {
```
Comment (ardatan) [inline packages/fets/src/createRouter.ts:156]: Why do we check `then`?

File: packages/fets/tests/router.spec.ts [hunk 2]
```diff
@@ -158,7 +159,32 @@ describe('Router', () => {
     });
     const response = await router.fetch('https://foo.com/foo?cat=dog&foo=bar&foo=baz&missing=');
     const json = await response.json();
-    expect(json).toMatchObject({ cat: 'dog', foo: ['bar', 'baz'], missing: [] });
+    expect(json).toMatchObject({ cat: 'dog', foo: ['bar', 'baz'], missing: undefined });
```
[the remark this answers is not in the document]
Comment (kylem123) [reply packages/fets/tests/router.spec.ts:162]: Yes, good spot, will address that too!

File: packages/fets/tests/router.spec.ts [hunk 3]
```diff
@@ -158,7 +159,32 @@ describe('Router', () => {
...
+  });
+  it('can handle missing Zod query params', async () => {
```
[the remark this answers is not in the document]
Comment (kylem123) [reply packages/fets/tests/router.spec.ts:164]: Have moved it to this file

File: packages/fets/tests/router.spec.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { z } from 'zod';
```
Comment (ardatan) [inline packages/fets/tests/router.spec.ts:1]: Could you remove this extra import?

**Comment by kylem123** [comment]
I believe I have addressed the issues you have mentioned. I have also added the result of the `yarn changeset`. Please let me know if anything else is needed :)

**Comment by kylem123** [comment]
Hi, wondered if you'd seen that this PR is ready to go. Would be good to be merged ASAP as we are currently having to run a yarn patch to handle this.

**Comment by ardatan** [comment]
https://github.com/ardatan/feTS/pull/223/files#r1238646645

This is still needed because `missing=` shouldn't be parsed as `undefined` but an empty string;
You can test the behavior in any JS env;
```ts
const url = new URL('http://dummy.com?missing=');
console.log(url.searchParams.get('missing'))
```

**Comment by kylem123** [comment]
> https://github.com/ardatan/feTS/pull/223/files#r1238646645
> 
> Unfortunately not :( This is still needed because `missing=` shouldn't be parsed as `undefined` but an empty string; You can test the behavior in any JS env;
> 
> ```ts
> const url = new URL('http://dummy.com?missing=');
> console.log(url.searchParams.get('missing'))
> ```

Hi, I believe I have resolved this with my latest commit, so it should now have the correct behaviour.

**Comment by ardatan** [comment]
Thanks @kylem123 ! I'll merge it after CI finishes!

**Comment by kylem123** [comment]
> Thanks @kylem123 ! I'll merge it after CI finishes!

Awesome, thank you!

**Pull request merged**